Assets/Scripts/Interaction/ChipInterfaceEditor.cs

```diff
@@ -4,577 +4,608 @@
+            DrawHandle(singnal.transform.position.y, handleState);
+        }
     }
-    if (selectedSignals[0] is OutputSignal) {
-      var signal = selectedSignals[0];
-      var pin = signal.inputPins[0];
-      wireType = pin.wireType;
+
+    ChipSignal GetSignalUnderMouse()
+    {
+        ChipSignal signalUnderMouse = null;
+        float nearestDst = float.MaxValue;
+
+        for (int i = 0; i < signals.Count; i++)
+        {
+            ChipSignal currentSignal = signals[i];
+            float handleY = currentSignal.transform.position.y;
+
+            Vector2 handleCentre = new Vector2(transform.position.x, handleY);
+            Vector2 mousePos = InputHelper.MouseWorldPos;
+
+            const float selectionBufferY = 0.1f;
+
+            float halfSizeX = handleSizeX;
+            float halfSizeY = (ScalingManager.handleSizeY + selectionBufferY) / 2f;
+            bool insideX = mousePos.x >= handleCentre.x - halfSizeX &&
+                           mousePos.x <= handleCentre.x + halfSizeX;
+            bool insideY = mousePos.y >= handleCentre.y - halfSizeY &&
+                           mousePos.y <= handleCentre.y + halfSizeY;
+
+            if (insideX && insideY)
+            {
+                float dst = Mathf.Abs(mousePos.y - handleY);
+                if (dst < nearestDst)
+                {
+                    nearestDst = dst;
+                    signalUnderMouse = currentSignal;
+                }
+            }
+        }
+        return signalUnderMouse;
     }
 
-    dragMouseStartY = InputHelper.MouseWorldPos.y;
-    if (selectedSignals.Count % 2 == 0) {
-      int indexA = Mathf.Max(0, selectedSignals.Count / 2 - 1);
-      int indexB = selectedSignals.Count / 2;
-      dragHandleStartY = (selectedSignals[indexA].transform.position.y +
-                          selectedSignals[indexB].transform.position.y) /
-                         2f;
-    } else {
-      dragHandleStartY =
-          selectedSignals[selectedSignals.Count / 2].transform.position.y;
+    // Select signal (starts dragging, shows rename field)
+    void SelectSignal(ChipSignal signalToDrag)
+    {
+        if (signalToDrag == null) return;
+        // Dragging
+        SelectAllSignalsInTheSameGroup(signalToDrag);
+
+        isDragging = true;
+
+
+        dragMouseStartY = InputHelper.MouseWorldPos.y;
+        if (selectedSignals.Count % 2 == 0)
+        {
+            int indexA = Mathf.Max(0, selectedSignals.Count / 2 - 1);
+            int indexB = selectedSignals.Count / 2;
+            dragHandleStartY = (selectedSignals[indexA].transform.position.y +
+                                selectedSignals[indexB].transform.position.y) /
+                               2f;
+        }
+        else
+        {
+            dragHandleStartY = selectedSignals[selectedSignals.Count / 2].transform.position.y;
+        }
+
+        PropertiesMenu.EnableUI(this, selectedSignals[0].signalName, IsGroup, selectedSignals[0].useTwosComplement,
```

Comment:
I belive `IsGroup` reports wrong values.

Shouldn't this be called something like:

```suggestion
        PropertiesMenu.EnableUI(this, selectedSignals[0].signalName, selectedSignals.Count() > 1, selectedSignals[0].useTwosComplement,
```